File: s3-source-connector/src/main/java/sourceconnector/domain/log/JSONLog.java
package sourceconnector.domain.log;

public class JSONLog implements Log {
  private final String payload;
  private final LogMetadata logMetadata;

  public JSONLog(String payload, LogMetadata logMetadata) {
    if (logMetadata == null) {
      throw new IllegalArgumentException("logMetadata cannot be null");
    }
    this.payload = payload;
    this.logMetadata = logMetadata;
  }

  public JSONLog withPayload(String newPayload) {
    return new JSONLog(newPayload, this.logMetadata);
  }

  @Override
  public LogMetadata getMetadata() {
    return this.logMetadata;
  }

  @Override
  public String get() {
    return this.payload;
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/log/JSONLog.kt
package sourceconnector.domain.log

class JSONLog(
  private val payload: String,
  override val metadata: LogMetadata
) : Log {

  fun withPayload(newPayload: String): JSONLog {
    return JSONLog(newPayload, this.metadata)
  }

  override fun get(): String {
    return this.payload
  }
}
